feathr_project/feathr/client.py

```diff
@@ -619,8 +620,15 @@ def materialize_features(self, settings: MaterializationSettings, execution_conf
             allow_materialize_non_agg_feature: Materializing non-aggregated features (the features without WindowAggTransformation) doesn't output meaningful results so it's by default set to False, but if you really want to materialize non-aggregated features, set this to True.
         """
         feature_list = settings.feature_names
-        if len(feature_list) > 0 and not self._valid_materialize_keys(feature_list):
-            raise RuntimeError(f"Invalid materialization features: {feature_list}, since they have different keys. Currently Feathr only supports materializing features of the same keys.")
+        if len(feature_list) > 0:
+            if 'anchor_list' in dir(self):
+                anchors = [anchor for anchor in self.anchor_list if isinstance(anchor.source, InputContext)]
+                anchor_names = set(feature.name  for anchor in anchors for feature in anchor.features)
```

Comment:
`anchor_names` sounds like a name of anchors, but from the context it sounds more like feature names in anchors? We should rename if that's the case